src/MassTransit.ApplicationInsights/ApplicationInsightsPublishFilter.cs

```diff
@@ -46,68 +57,23 @@ public void Probe(ProbeContext context)
 
 		public async Task Send(T context, IPipe<T> next)
 		{
-			var requestTelemetry = new DependencyTelemetry()
+			var contextType = context.GetType();
+			var messageType = contextType.GetGenericArguments().FirstOrDefault()?.FullName ?? "Unknown";

```

Comment:
This seems slow and expensive, why isn't this filter using `PublishContext<T>` instead of `T`?

Also, perhaps a cache similar to the TypeMetadataCache<T>, which will cache the TypeMetadataCache<T>.ShortName for the various publish/consume/send context generics, to avoid the performance hit every call?

Reply:
> why isn't this filter using PublishContext<T> instead of T?

That would be perfect, cause during exception we can log whole "invalid" object.
But I don't know how to register most generic Type into pipeline. I have tried with 'object', but I get error with invalid message type:
"System.ArgumentException: 'Messages types must not be System types: System.Object'"
Do you have any clue?
Code:
`pipeConfigurator.AddPipeSpecification(new ApplicationInsightsPublishSpecificationClass<object>(telemetryClient, telemetryHeaderRootKey, telemetryHeaderParentKey, null));`

PS. I have took it from this from merged pull request ;) https://github.com/MassTransit/MassTransit/pull/1264